api/bazel/api_build_system.bzl

```diff
@@ -181,3 +145,50 @@ def api_go_test(name, size, importpath, srcs = [], deps = []):
         importpath = importpath,
         deps = deps,
     )
+
+_GO_BAZEL_RULE_MAPPING = {
+    "@opencensus_proto//opencensus/proto/trace/v1:trace_proto": "@opencensus_proto//opencensus/proto/trace/v1:trace_proto_go",
+    "@opencensus_proto//opencensus/proto/trace/v1:trace_config_proto": "@opencensus_proto//opencensus/proto/trace/v1:trace_and_config_proto_go",
+    "@com_google_googleapis//google/api/expr/v1alpha1:syntax_proto": "@com_google_googleapis//google/api/expr/v1alpha1:cel_go_proto",
+}
+
+def go_proto_mapping(dep):
+    mapped = _GO_BAZEL_RULE_MAPPING.get(dep)
+    if mapped == None:
+        return _Suffix("@" + Label(dep).workspace_name + "//" + Label(dep).package + ":" + Label(dep).name, _GO_PROTO_SUFFIX)
+    return mapped
+
+def api_proto_package(name = "pkg", srcs = [], deps = [], has_services = False, visibility = ["//visibility:public"]):
+    if srcs == []:
+        srcs = native.glob(["*.proto"])
+
+    native.proto_library(
+        name = name,
+        srcs = srcs,
+        deps = deps + _COMMON_PROTO_DEPS,
+        visibility = visibility,
+    )
+
+    compilers = ["@io_bazel_rules_go//proto:go_proto", "//bazel:pgv_plugin_go"]
+    if has_services:
+        compilers = ["@io_bazel_rules_go//proto:go_grpc", "//bazel:pgv_plugin_go"]
```

Comment:
Does this mean that `go_proto` and `go_grpc` will not be separate targets anymore? So users no longer have to refer to `*_go_grpc` and can always just use `*_go_proto`?

Reply:
Yes, it is generally the same go package for proto and gRPC in OSS. Internal rules might still create two different rules if necessary.